modules/xml-sitemap/sitemap.php

```diff
@@ -0,0 +1,595 @@
+				$news['news:publication']['news:language'] = get_blog_lang_code();
+			}
+			$news['news:publication_date'] = jetpack_w3cdate_from_mysql( $post->post_date_gmt );
+			$news['news:title']            = get_the_title_rss();
+			if ( $post->keywords ) {
+				$news['news:keywords'] = html_entity_decode( ent2ncr( $post->keywords ), ENT_HTML5 );
+			}
+			$url['news:news'] = $news;
+
+			// Add image to sitemap
+			if ( current_theme_supports( 'post-thumbnails' ) && has_post_thumbnail( $post->ID ) ) {
+				$post_thumbnail_id  = get_post_thumbnail_id( $post->ID );
+				$post_thumbnail_src = wp_get_attachment_image_src( $post_thumbnail_id );
+				if ( $post_thumbnail_src ) {
+					$url['image:image'] = array ( 'image:loc' => esc_url( $post_thumbnail_src[0] ) );
+				}
+			}
+
+			/**
+			 * Filter associative array with data to build <url> node and its descendants for current post in news sitemap.
+			 *
+			 * @since 3.9
+			 *
+			 * @param array $url Data to build parent and children nodes for current post.
+			 * @param int $post_id Current post ID.
+			 */
+			$url = apply_filters( 'jetpack_sitemap_news_sitemap_item', $url, $post );
+
+			if ( empty( $url ) ) {
+				continue;
+			}
+
+			jetpack_print_sitemap_item( $url );
+		endforeach;
+		?>
+	</urlset>
+	<?php
+	$xml = ob_get_contents();
+	ob_end_clean();
+	if ( ! empty( $xml ) ) {
+		set_transient( 'jetpack_news_sitemap', $xml, DAY_IN_SECONDS );
+		echo $xml;
+	}
+
+	die();
+}
+
+/**
+ * Absolute URL of the current blog's sitemap
+ *
+ * @return string sitemap URL
+ */
+function jetpack_sitemap_uri() {
+	/**
+	 * Filter sitemap URL relative to home URL.
+	 *
+	 * @since 3.9
+	 *
+	 * @param string $sitemap_url Sitemap URL.
+	 */
+	return apply_filters( 'jetpack_sitemap_location', home_url( '/sitemap.xml' ) );
+}
+
+/**
+ * Absolute URL of the current blog's news sitemap
+ */
+function jetpack_news_sitemap_uri() {
+	/**
+	 * Filter news sitemap URL relative to home URL.
+	 *
+	 * @since 3.9
+	 *
+	 * @param string $news_sitemap_url News sitemap URL.
+	 */
+	return apply_filters( 'jetpack_news_sitemap_location', home_url( '/news-sitemap.xml' ) );
+}
+
+/**
+ * Output the master sitemap URLs for the current blog context
+ */
+function jetpack_sitemap_discovery() {
+	if ( ! defined( 'JETPACK_SKIP_DEFAULT_SITEMAP' ) || true !== JETPACK_SKIP_DEFAULT_SITEMAP ) {
+		echo 'Sitemap: ' . esc_url( jetpack_sitemap_uri() ) . PHP_EOL;
+	}
+
+	if ( ! defined( 'JETPACK_SKIP_DEFAULT_NEWS_SITEMAP' ) || true !== JETPACK_SKIP_DEFAULT_NEWS_SITEMAP ) {
+		echo 'Sitemap: ' . esc_url( jetpack_news_sitemap_uri() ) . PHP_EOL . PHP_EOL;
+	}
+}
+
+/**
+ * Clear the sitemap cache when a sitemap action has changed
+ *
+ * @param int $post_id unique post identifier. not used.
+ */
+function jetpack_sitemap_handle_update( $post_id ) {
+	delete_transient( 'jetpack_sitemap' );
+	delete_transient( 'jetpack_news_sitemap' );
+}
+
+if ( ! function_exists( 'is_publicly_available' ) || is_publicly_available() ) {
```

Comment:
We have a `check_privacy` function in Jetpack:
https://github.com/Automattic/jetpack/blob/master/class.jetpack.php#L5410

Maybe we should use it in this case?